src/main/java/com/arpnetworking/kairosdb/aggregators/HistogramFilterAggregator.java

```diff
@@ -0,0 +1,281 @@
+        _filterinc = inclusion;
+    }
+
+    @Override
+    public DataPointGroup aggregate(final DataPointGroup dataPointGroup) {
+        return new HistogramFilterDataPointAggregator(dataPointGroup);
+    }
+
+    @Override
+    public boolean canAggregate(final String groupType) {
+        return HistogramDataPointFactory.GROUP_TYPE.equals(groupType);
+    }
+
+    @Override
+    public String getAggregatedGroupType(final String groupType) {
+        return HistogramDataPointFactory.GROUP_TYPE;
+    }
+
+    private static double truncate(final double val) {
+        final long mask = 0xffffe00000000000L;
+        return Double.longBitsToDouble(Double.doubleToRawLongBits(val) & mask);
+    }
+
+    /**
+     * Gives an inclusive bound of the bin that the passed in value will be placed in.
+     * If the value is positive, it will be an inclusive upper bound, and if it is negative,
+     * it will be an inclusive lower bound.
+     *
+     * @param val the value whose bucket bound will be calculated
+     * @return the inclusive upper or lower bound of the bin
+     */
+    private static double binInclusiveBound(final double val) {
+        long bound = Double.doubleToLongBits(val);
+        bound >>= 45;
+        bound += 1;
+        bound <<= 45;
+        bound -= 1;
+
+        return Double.longBitsToDouble(bound);
+    }
+
+    private static boolean isNegative(final double value) {
+        return Double.doubleToLongBits(value) < 0;
+    }
+
+    private class HistogramFilterDataPointAggregator extends AggregatedDataPointGroupWrapper {
+        HistogramFilterDataPointAggregator(final DataPointGroup innerDataPointGroup) {
+            super(innerDataPointGroup);
+        }
+
+        public boolean hasNext() {
+            return currentDataPoint != null;
+        }
+
+        public DataPoint next() {
+            final DataPoint dp = currentDataPoint;
+            final long timeStamp = dp.getTimestamp();
+            final TreeMap<Double, Integer> filtered = Maps.newTreeMap();
+            double min = Double.MAX_VALUE;
+            double max = -Double.MAX_VALUE;
+            double sum = 0;
+            long count = 0;
+
+            if (dp instanceof HistogramDataPoint) {
+                final HistogramDataPoint hist = (HistogramDataPoint) dp;
+
+                for (final Map.Entry<Double, Integer> entry : hist.getMap().entrySet()) {
+                    final double binValue = entry.getKey();
+                    final int binCount = entry.getValue();
+                    if (!shouldDiscard(binValue)) {
+                        filtered.put(binValue, binCount);
+                        min = Math.min(min, binValue);
+                        max = Math.max(max, binValue);
+                        sum += binValue * binCount;
+                        count += binCount;
+                    }
+                }
+            }
+
+            final double mean = sum / count;
+            if (filtered.size() <= 0) {
```

Comment:
Trying to understand why `mean` is not defaulted but the inputs to it are in the case of filtered being empty. Also although filtered being empty implies count is empty, filtered being non-empty does not necessarily (although likely does practically) mean that count is non-zero.